src/core/mesh/qgsmeshdataprovider.h

```diff
@@ -475,22 +575,43 @@ class CORE_EXPORT QgsMeshDatasetSourceInterface SIP_ABSTRACT
 
     /**
      * \brief Returns vector/scalar value associated with the index from the dataset
-     * To read multiple continuous values, use QgsMeshDatasetSourceInterface::datasetValues()
+     * To read multiple continuous values, use datasetValues()
      *
      * See QgsMeshDatasetMetadata::isVector() or QgsMeshDataBlock::type()
      * to check if the returned value is vector or scalar
+     *
+     * Returns invalid value for DataOnVolumes
+     *
+     * \see datasetValues
      */
     virtual QgsMeshDatasetValue datasetValue( QgsMeshDatasetIndex index, int valueIndex ) const = 0;
 
     /**
      * \brief Returns N vector/scalar values from the index from the dataset
      *
-     * See QgsMeshDatasetMetadata::isVector() to check if the returned value is vector or scalar
+     * See QgsMeshDatasetMetadata::isVector() or QgsMeshDataBlock::type()
+     * to check if the returned value is vector or scalar
+     *
+     * Returns invalid block for DataOnVolumes. Use QgsMeshLayerUtils::datasetValues() if you
+     * need block for any type of data typ
```

Comment:
s/typ/type/

Maybe good to add that QgsMeshLayerUtils::datasetValues() would apply specified averaging method to a 3D dataset to convert it to 2D dataset.